sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/metadata/datasource/dialect/MySQLDataSourceMetaData.java

```diff
@@ -32,24 +35,39 @@
  */
 @Getter
 public final class MySQLDataSourceMetaData implements DataSourceMetaData {
-    
+
     private static final int DEFAULT_PORT = 3306;
-    
+
     private final String hostName;
-    
+
     private final int port;
-    
+
     private final String schemaName;
-    
+
+    private final String catalog;
+
     private final Pattern pattern = Pattern.compile("jdbc:(mysql|mysqlx)(:loadbalance|:replication)?:(\\w*:)?//([\\w\\-\\.]+):?([0-9]*)/([\\w\\-]+);?\\S*", Pattern.CASE_INSENSITIVE);
-    
+
     public MySQLDataSourceMetaData(final String url) {
```

Comment:
Why does it exist if `MySQLDataSourceMetaData(final DataSourceInfo dataSourceInfo)` is added?

Reply:
did it